File: SchoolManagement.Web/ClientApp/src/app/models/report-filter.ts
import { Guid } from "guid-typescript";
export interface ReportFilter {
  FromDate?: Date;
  ToDate?: Date;
  Session?: number;
  ClassIds?: Array<Guid>;
  SectionIds?: Array<Guid>;
  FeeTypeIds?: Array<Guid>;
  ClassId?: Guid;
  SectionId?: Guid;
  FeeTypeId?: Guid;
  StudentName?: string;
  AdmissionNo?: string;
}

File: SchoolManagement.Model/ReportFilter.cs
using System;
using System.Collections.Generic;

namespace SchoolManagement.Model
{
    public class ReportFilter
    {
        public List<Guid> ClassIds { get; set; }
        public List<Guid> FeeTypeIds { get; set; }
        public List<Guid> SectionIds { get; set; }
        public int Session { get; set; }
        public string AdmissionNo { get; set; }
        public string StudentName { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
        public Guid? FeeTypeId { get; set; }
        public Guid? ClassId { get; set; }
        public Guid? SectionId { get; set; }
    }
}